### PR TITLE
[CIT-577] Update postinst script of tedge-agent

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes
By this change, `tedge-agent` user can run `sudo /etc/tedge/sm-plugins/[plugin]`.

Restriction:
- Due to security issue, **we force a plugin name must start with [a-z].** If we use wildcard like `/etc/tedge/sm-plugins/*`, hacker can execute `/etc/tedge/sm-plugins/../../anything`.
- sudoers doesn't accept regrex. See man page: https://www.sudo.ws/man/sudoers.man.html#Wildcards

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by didier-wenzek — APPROVED** [review]
Approved.

One might also accept other characters (say [a-zA-Z0-9]) as the first one of the command name. Admittedly unusual on unix, but allowed.

File: configuration/debian/tedge_agent/postinst [hunk 1]
```diff
@@ -13,10 +13,10 @@ if ! getent passwd tedge-agent > /dev/null; then
 fi
 
 ### Create file in /etc/sudoers.d directory
-echo "%tedge-agent   ALL = (ALL) NOPASSWD: /etc/tedge/sm-plugins/apt" > /etc/sudoers.d/tedge-agent
+echo "%tedge-agent   ALL = (ALL) NOPASSWD: /etc/tedge/sm-plugins/[a-z]*" > /etc/sudoers.d/tedge-agent
 
 if [ -f "/etc/sudoers.d/010_pi-nopasswd" ]; then
-    echo "%tedge-agent   ALL = (ALL) NOPASSWD: /etc/tedge/sm-plugins/apt" > /etc/sudoers.d/tedge-agent-nopasswd
+    echo "%tedge-agent   ALL = (ALL) NOPASSWD: /etc/tedge/sm-plugins/[a-z]*" > /etc/sudoers.d/tedge-agent-nopasswd
```
Comment (PradeepKiruvale) [inline configuration/debian/tedge_agent/postinst:19]: What if there are numbers, upper case letters, and special characters in the plugin name?

Comment (albinsuresh) [reply configuration/debian/tedge_agent/postinst:19]: Uppercase and numbers, definitely. Even if we decide to support special characters, please keep it minimal to some common chars in names like `-` and `_`.

**Review by albinsuresh — COMMENTED** [review]
You might wanna add a pysys test to test this multi-plugin support.

File: configuration/debian/tedge_agent/postinst (hunk 1 above)
Comment (rina23q) [reply configuration/debian/tedge_agent/postinst:19]: `[a-z]` limitation applies only the first character. Afterwards, user can use any character, this is `*` means. I agree `[a-zA-Z0-9]` but not convinced to add special characters.

**Review by abelikt — APPROVED** [review]
Looks good for me.
Wondering if we ever can make this secure.

Hackers could aso use /etc/tedge/sm-plugins/**a**/../../anything afterwards, right?

**Comment by rina23q** [comment]
> Hackers could aso use /etc/tedge/sm-plugins/**a**/../../anything afterwards, right?

I tested, actually in this case, you are asked to input the password of `tedge-agent`. So, seems it's safe.

No password is asked in this case: `/etc/tedge/sm-plugins/**a**/../apt`.

**Review by PradeepKiruvale — APPROVED** [review]

**Pull request merged**